src/stores/AuthStore.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+import type RootStore from './RootStore';
+
+import DAOApi from 'brewskey.js-api';
+import authApi from '../authApi';
+import { action, autorun, computed, observable } from 'mobx';
+
+const setDAOHeaders = (token: string) => {
+  const daoHeaders = DAOApi.getHeaders().filter(
+    (header: { name: string, value: string }): boolean =>
+      header.name !== 'Authorization' && header.name !== 'timezoneOffset',
+  );
+
+  DAOApi.setHeaders([
+    ...daoHeaders,
+    {
+      name: 'timezoneOffset',
+      value: new Date().getTimezoneOffset().toString(),
+    },
+    {
+      name: 'Authorization',
+      value: `Bearer ${token}`,
+    },
+  ]);
+};
+
+class AuthStore {
+  store: RootStore;
+
+  @observable authState = new Map();
+
+  constructor(rootStore) {
+    this.store = rootStore;
+
+    // todo move from constructor
+    autorun(() => {
```

Comment:
not sure it is an appropriate use case for `autorun`. Basically, what I want is: whenever we get some error with status 401, we  clean up authState and redirect to login screen (should  redirection  logic be in the store?)

Reply:
It should call an action to update the data... or since this is mobx maybe just mutate the data here is fine.